test/integration/apiserver/flowcontrol/fight_test.go

```diff
@@ -119,25 +117,22 @@ func (ft *fightTest) countWrite(fs *flowcontrol.FlowSchema) {
 }
 
 func (ft *fightTest) createController(invert bool, i int) {
-	fieldMgr := fmt.Sprintf("testController%d%v", i, invert)
+	fieldMgr := fmt.Sprintf("testController-inverting-is-%v", invert)
+	name := fmt.Sprintf("testController-%d-%v", i, invert)
 	myConfig := rest.CopyConfig(ft.loopbackConfig)
-	myConfig = rest.AddUserAgent(myConfig, fieldMgr)
+	myConfig = rest.AddUserAgent(myConfig, name)
 	myClientset := clientset.NewForConfigOrDie(myConfig)
 	fcIfc := myClientset.FlowcontrolV1beta1()
 	informerFactory := informers.NewSharedInformerFactory(myClientset, 0)
-	foundToDangling := func(found bool) bool { return !found }
-	if invert {
-		foundToDangling = func(found bool) bool { return found }
-	}
 	ctlr := utilfc.NewTestable(utilfc.TestableConfig{
-		Name:                   fieldMgr,
-		FoundToDangling:        foundToDangling,
+		Name:                   name,
```

Comment:
`func (cfgCtlr *configController) digestConfigObjects` puts `cfgCtlr.asFieldManager` in the `metav1.PatchOptions` used there.  This is something that will differ between teams but not between individuals on a team.  To help understand what is logged, it is helpful to also have an identifier that distinguishes between individuals.

Reply:
> func (cfgCtlr *configController) digestConfigObjects puts cfgCtlr.asFieldManager in the metav1.PatchOptions used there.

I'm aware of that. But in all production usecase it is set to predefined constant anyway:
https://github.com/kubernetes/kubernetes/blob/3f09686958f00420ae1608ae9997ceb4d06e7006/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_filter.go#L39

So I don't see a good reason for having more complicated production code just to have a bit better logs in tests.

@deads2k - if you disagree with the above

Reply:
Yes, that constant is the point here.  If and when we change the controller behavior in a way that would cause a fight in a mixed population, the newer controllers will have to use a newer constant.  See how it includes a version number?  So that is why the two teams in the test use different field manager values, and why the field manager value is the same among all members of the same team.

Reply:
> If and when we change the controller behavior

The "If" part is interesting. If we change that at some point, we can readd that. But (unless you already know we will change in in the upcoming release), I suggest removing it - we can always readd this.

Reply:
We have discussed how we would handle future behavior changes, and testing that behavior is what this test is about.  In order to test how a fight will proceed in a mixed population of apiservers, we need to actually create a mixed population of apiservers.  So what you see here is our plan for what that will look like.

Reply:
But you're not exercising the existence of the "asFieldManager" field at all - you don't have to change anything in the test if ou remove it.
The only thing that changes is the logs in the test.
And that to me doesn't justify complicating production code.

Reply:
I do not understand that comment.  There is a field manager string that goes into the PatchOptions.  In the scenario we want to test --- a mixed population of current controllers and future controllers --- that field manager string will differ between the two sub-populations.  So we need the field manager string to be a parameter in some way.

Reply:
I guess now I don't understand the comment.

Yes - the field manager string goes into PatchOptions. Which then translates to the content of managed fields.
But the code doesn't rely on that anywhere unless I'm missing something.

All the reconciliation is based on what objects currently exist and what the controller believes that should exist. We don't look at the values of ManagedFields at all - it's purely informative.

So in other word - no matter to what value you set that to currently, it won't really matter for correctness of the code.
[And in production code we always hardcode it to that constant.]

So as of now, the only reason for allowing to modify it (i.e. having AsFieldManager in the config) is basically this test.
And even this test doesn't rely on this being set anywhere - it's only useful for debugging purposes as of now.

So that's why I'm saying that complicating production code for no real gain other than easier debugging of tests is not worth it.

Reply:
Yes, this test code does not read the field manager.  However, the field manager is, or may be, relevant to server-side behavior.  If we take out the difference in field manager from this test, that is asserting that we know that the field manager makes no difference to server-side behavior.